pkg/server/server.go

```diff
@@ -445,8 +445,9 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 				})
 			if err := backend.Send(pkt); err != nil {
 				klog.ErrorS(err, "DIAL_REQ to Backend failed", "serverID", s.serverID)
+			} else {
+				klog.V(5).Infoln("DIAL_REQ sent to backend") // got this. but backend didn't receive anything.
```

Comment:
This makes me wonder if in case of error, we want to do nothing, as we currently do, or we want to "break" or something. If we want to break, then the else is not needed...

If we break here, we close the conn. I'm not sure if this is what we want, but unsure if keeping it open if the DIAL_REQ failed and is not retried is a state we want to be either... What do you think?

Reply:
@andrewsykim What do you think?

Reply:
I definitely agree to revisit this logic and make sure we're returning correctly in the right places. I have to give it more thought.